File: maddie-carlson/day-08/who-likes-this.js
function likes(names) {
  // takes an array of names and returns a string showing who likes what, like how Facebook does it
  // returns a string detailing the likes
  // can fail if names is the wrong type or the types inside are not strings

  //check for empty array
  if (names.length === 0) {
    return "no one likes this"
  }

  let likes = ""

  //add names based on array length
  //takes at most 3 names, or two names if it's four or more, instead having a number

  if (names.length === 1) {
    likes = names[0] + " likes this"
  } else if (names.length === 2) {
    likes = names[0] + " and " + names[1] + " like this"
  } else if (names.length === 3) {
    likes = names[0] + ", " + names[1] + " and " + names[2] + " like this" //no oxford comma :(
  } else { //more than 3
    likes = names[0] + ", " + names[1] + " and " + String(names.length - 2) + " others like this" //just add to the count
  }

  return likes
}
/*
  Ember's Feedback:
  - Great job!
  - Use === it will save you headaches down the road
*/
// Alternative solution
function likes(names) {
  switch (names.length) {
    case 0:
      return 'no one likes this';
    case 1:
      return `${names[0]} likes this`;
    case 2:
      return `${names[0]} and ${names[1]} like this`;
    case 3:
      return `${names[0]}, ${names[1]} and ${names[2]} like this`;
    default:
      return `${names[0]}, ${names[1]} and ${names.length - 2} others like this`;
  }
}

File: nathan-wilson/day-38/who_likes_this.py
def likes(names):
    length = len(names)
    
    if length == 0:
        return "no one likes this"
    elif length == 1:
        return f"{names[0]} likes this"
    elif length == 2:
        return f"{names[0]} and {names[1]} like this"
    elif length == 3:
        return f"{names[0]}, {names[1]} and {names[2]} like this"
    else:
        return f"{names[0]}, {names[1]} and {length-2} others like this"
    
# array of names

# string

# ['Jacob', 'Alex']), 'Jacob and Alex like this'

# I could return the incorrect string or have a type error

# if length is 0 or 1 2 3 4+ handle the specific format of the string

# I can do this!